content/shared/influxdb3-processing-engine.md

```diff
@@ -0,0 +1,77 @@
+# Processing engine
+
+The Processing engine is an embedded Python virtual machine that runs inside the InfluxDB database. It executes Python code in response to database events without requiring external application servers or middleware.
+
+## How it works
+
+### Architecture
+
+The Processing engine runs Python code directly within the InfluxDB process. This design provides high performance and direct access to database resources.
+
+- **Embedded execution**: Code runs in the same process space as the database
+- **Direct data access**: Zero-copy access to data where possible
+- **Event-driven**: Responds to database writes, scheduled events, and HTTP requests
+- **Isolated contexts**: Maintains separation between different plugin executions
+
```

Comment:
You could add a bullet that mentions access to the Distinct values and Last values caches, which are very important features.